driver-core/src/main/com/mongodb/internal/connection/LoadBalancedCluster.java

```diff
@@ -111,7 +132,7 @@ public void initialize(final MongoException initializationException) {
 
                 @Override
                 public ClusterType getClusterType() {
-                    return initialized ? ClusterType.LOAD_BALANCED : ClusterType.UNKNOWN;
+                    return initializationCompleted ? ClusterType.LOAD_BALANCED : ClusterType.UNKNOWN;
                 }
             });
             dnsSrvRecordMonitor.start();
```

Comment:
I would rather not make this change independently of making a similar change in all other Cluster implementations.

Reply:
[optional]

The `start` method does not need to be in the interface `Cluster` (theoretically it may not be needed by some implementations), and therefore should not be there. Thus, it can be introduced for `LoadBalancedCluster` without introducing it for other implementations.

With the above in mind, the only argument agains introducing `LoadBalancedCluster.start` I can imagine would be: _"its siblings start threads in constructors, so even if the approach is unsound, by continuing to use it we are making the code style more consistent"_. For such an imaginary argument I'd respond that while consistency of the code style is important, using sound approaches is more important.

I presented all arguments I currently have with regard to this concern, and I am marking this proposal as optional, thus signifying that I will make my peace with either of the decisions.

Reply:
I'm going to leave it as is and open a separate ticket.  I agree there is a value judgement here, and I guess I'm coming down on the consistency side.

But I agree that it's a better approach to introduce a start/init method, so I will open a separate ticket to address that technical debt.

Reply:
👍

Reply:
JAVA-4175